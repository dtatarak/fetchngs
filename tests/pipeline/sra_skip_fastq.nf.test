nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"
    options "-profile test,docker"
    tag "pipeline"

    test("SRA Skip FASTQ") {

        when {
            params {
                outdir              = "$outputDir"
                skip_fastq_download = true
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(path("$outputDir/pipeline_info/software_versions.yml")).match("software_versions") },
                { assert snapshot(path("$outputDir/metadata/").list()).match("metadata") },
                { assert snapshot(path("$outputDir/samplesheet/").list()).match("samplesheet") },
                { assert !new File("$outputDir/fastq/DRX024467_DRR026872.fastq.gz").exists() },
                { assert !new File("$outputDir/fastq/DRX026011_DRR028935_1.fastq.gz").exists() },
                { assert !new File("$outputDir/fastq/DRX026011_DRR028935_2.fastq.gz").exists() },
                { assert !new File("$outputDir/fastq/ERX1188904_ERR1109373_1.fastq.gz").exists() },
                { assert !new File("$outputDir/fastq/ERX1188904_ERR1109373_2.fastq.gz").exists() },
                { assert !new File("$outputDir/fastq/ERX1234253_ERR1160846.fastq.gz").exists() },
                { assert !new File("$outputDir/fastq/SRX10940790_SRR14593545_1.fastq.gz").exists() },
                { assert !new File("$outputDir/fastq/SRX10940790_SRR14593545_2.fastq.gz").exists() },
                { assert !new File("$outputDir/fastq/SRX11047067_SRR14709033.fastq.gz").exists() },
                { assert !new File("$outputDir/fastq/SRX6725035_SRR9984183.fastq.gz").exists() },
                { assert !new File("$outputDir/fastq/SRX9315476_SRR12848126_1.fastq.gz").exists() },
                { assert !new File("$outputDir/fastq/SRX9315476_SRR12848126_2.fastq.gz").exists() },
                { assert !new File("$outputDir/fastq/SRX9626017_SRR13191702_1.fastq.gz").exists() },
                { assert !new File("$outputDir/fastq/SRX9626017_SRR13191702_2.fastq.gz").exists() }
            )
        }

    }

}
