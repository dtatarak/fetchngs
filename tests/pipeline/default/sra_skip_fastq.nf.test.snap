{
    "metadata": {
        "content": [
            "DRR026872.runinfo_ftp.tsv:md5,5620768b65ecb4067b7547a49e07b67b",
            "DRR028935.runinfo_ftp.tsv:md5,7f248072a55d5f729c4827937625a05b",
            "ERR1109373.runinfo_ftp.tsv:md5,f7cd42805764f783388c2686fe2a3f95",
            "ERR1160846.runinfo_ftp.tsv:md5,36257a2c851d10d61a97581f3496529d",
            "SRR12848126.runinfo_ftp.tsv:md5,ce5380eceefba1210df0cefe321be62a",
            "SRR13191702.runinfo_ftp.tsv:md5,37f87c5665e65cece1bc5e2b529e3b3b",
            "SRR14593545.runinfo_ftp.tsv:md5,36a0fe8c5d3917f105e4cbbb7a720d7d",
            "SRR14709033.runinfo_ftp.tsv:md5,8f06fa2e4708318dbe8e9505000ea919",
            "SRR9984183.runinfo_ftp.tsv:md5,72047c09c02782932672862e8adf3fa5"
        ],
        "timestamp": "2023-04-27T16:46:30+0000"
    },
    "samplesheet": {
        "content": [
            "id_mappings.csv:md5,660db7a31a1c775cabad5f9b21addc06",
            "multiqc_config.yml:md5,d01bd71fb5b9901c80166fba0637e430",
            "samplesheet.csv:md5,6b59e47a9fffdde864907419d0c4ad04"
        ],
        "timestamp": "2023-04-27T16:46:30+0000"
    },
    "software_versions": {
        "content": [
            "software_versions.yml:md5,d9a2185a464cbd9c0f576dad2ccea2cc"
        ],
        "timestamp": "2023-04-27T16:46:30+0000"
    }
}